lib/active_merchant/billing/gateways/moneris.rb

```diff
@@ -324,10 +324,14 @@ def cvd_element(cvd_value)
         element
       end
 
+      def cof_details_present?(parameters)
+        parameters[:issuer_id] && parameters[:payment_indicator] && parameters[:payment_information]
```

Comment:
Just to verify (since this is not true for Worldpay), are all three fields actually required in both initial and subsequent cof txns?

Reply:
yes, @curiousepic, all three are required for first and subsequent cof transactions.